File: Documentation/snippets/python/string_reversal.py
def reverse_string(s: str) -> str:
    """Reverses a given string.

    Args:
        s: A string to be reversed.

    Returns:
        A string that is the reverse of the input.

    Raises:
        ValueError: If the input is not a string.
    """
    if not isinstance(s, str):
        raise ValueError("Input must be a string.")

    return s[::-1]

# Example usage:
try:
    original_string = "hello"
    reversed_string = reverse_string(original_string)
    print(f"Reversed string: {reversed_string}")
except ValueError as e:
    print(e)

#####################################################################

def reverse_string(s: str) -> str:
    """Reverses a given string without using standard libraries.

    Args:
        s: A string to be reversed.

    Returns:
        A string that is the reverse of the input.
    """
    reversed_string = ""
    for i in range(len(s) - 1, -1, -1):
        reversed_string += s[i]
    return reversed_string

# Example usage:
original_string = "hello"
reversed_string = reverse_string(original_string)
print(f"Reversed string: {reversed_string}")

File: Documentation/snippets/rust/string_reversal.rs
/// Reverses a given string.
///
/// # Arguments
///
/// * `s` - A string slice that holds the string to be reversed.
///
/// # Returns
///
/// * A new `String` that is the reverse of `s`.
///
/// # Errors
///
/// * This function will return an error if the string is empty.
fn reverse_string(s: &str) -> Result<String, &'static str> {
    if s.is_empty() {
        return Err("Input string must not be empty.");
    }

    Ok(s.chars().rev().collect::<String>())
}

// Example usage:
fn main() {
    let original_string = "hello";
    match reverse_string(original_string) {
        Ok(reversed) => println!("Reversed string: {}", reversed),
        Err(e) => println!("Error: {}", e),
    }
}


fn reverse_string(s: &mut [u8]) {
    let len = s.len();
    for i in 0..len / 2 {
        s.swap(i, len - i - 1);
    }
}

// Example usage:
fn main() {
    let mut original_string = String::from("hello").into_bytes();
    reverse_string(&mut original_string);
    let reversed_string = String::from_utf8(original_string).expect("Invalid UTF-8");
    println!("Reversed string: {}", reversed_string);
}
